### PR TITLE
chore (ui-react): propagate isDisabled to Clear button in SearchField

### LINKED ISSUE #3684: SearchField. Disable clear icon for disabled input
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Primitive components

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

_No response_

### Please describe your bug.

SearchField component has clear icon, which appears as soon as some text is entered into the input.
If SearchField is disabled, icon is still present, is focusable and clickable. Cursor looks like pointer.

### What's the expected behaviour?

Clear icon should be either removed or deactivated for disabled SearchField.

### Help us reproduce the bug!

searchField bug amplify

### Code Snippet

_No response_

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by helmax-y**
#### Issue 
Fixes #3684

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/SearchField/SearchField.tsx [hunk 1]
```diff
@@ -63,22 +63,24 @@ const SearchFieldPrimitive: Primitive<SearchFieldProps, 'input'> = (
     </FieldGroupIcon>
   ) : undefined;
 
+  const ClearButton = !isDisabled ? (
+    <FieldClearButton
+      ariaLabel={clearButtonLabel}
+      excludeFromTabOrder
+      isVisible={strHasLength(composedValue)}
```
Comment (helmax-y) [inline packages/react/src/primitives/SearchField/SearchField.tsx:70]: if it's more logical, I can rewrite using this isVisible prop
```suggestion
      isVisible={!isDisabled && strHasLength(composedValue)}
```

Comment (ioanabrooks) [reply packages/react/src/primitives/SearchField/SearchField.tsx:70]: I prefer the isVisible prop implementation, makes it easier to read for me. Want to hear what others think.

File: packages/react/src/primitives/SearchField/SearchField.tsx [hunk 2]
```diff
@@ -63,22 +63,24 @@ const SearchFieldPrimitive: Primitive<SearchFieldProps, 'input'> = (
...
-        <FieldClearButton
-          ariaLabel={clearButtonLabel}
-          excludeFromTabOrder
-          isVisible={strHasLength(composedValue)}
-          onClick={onClearHandler}
-          size={size}
-          variation="link"
-        />
-      }
+      innerEndComponent={ClearButton}
```
Comment (ioanabrooks) [inline packages/react/src/primitives/SearchField/SearchField.tsx:83]: Please write a unit test for the changes to prevent regressions.

**Review by ioanabrooks — COMMENTED** [review]
Thank you for contributing!

**Review by zchenwei — COMMENTED** [review]
LGTM. But could you unit test your changes before I approve it?

**Comment by helmax-y** [comment]
@zchenwei @ioanabrooks test was added

**Review by ioanabrooks — APPROVED** [review]

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Review by reesscot — APPROVED** [review]
Thank you for your contribution!

**Review by zchenwei — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Comment by offlineprogrammer** [comment]
@helmax-y Thank you for your contribution! I would like to follow up with you on this. can you please connect with me on [Twitter](https://twitter.com/_Mo_Malaka_) or [LinkedIn](https://www.linkedin.com/in/momalaka/)

**Pull request merged**